qualtran/bloqs/block_encoding.py

```diff
@@ -46,13 +48,62 @@
     Soquet,
     SoquetT,
 )
+from qualtran.bloqs.basic_gates import Toffoli
+from qualtran.bloqs.multi_control_multi_target_pauli import MultiControlPauli
 from qualtran.bloqs.select_and_prepare import PrepareOracle, SelectOracle
 from qualtran.bloqs.util_bloqs import Partition
+from qualtran.drawing import Circle, WireSymbol
+
+if TYPE_CHECKING:
+    from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+
+
+@attrs.frozen
+class Reflection(Bloq):
+    bitsizes: Tuple[int]
+    cvs: Tuple[int]
+
+    def __attrs_post_init__(self):
+        if len(self.bitsizes) != len(self.cvs):
+            raise ValueError(
+                f"cvs must be same length as bitsizes: {len(self.cvs)} vs {len(self.bitsizes)}"
+            )
+
+    def pretty_name(self) -> str:
+        return 'Refl'
```

Comment:
should this be `short_name`? I know we don't really have guidance on this

Reply:
Done.